Week3/homework/step2-6.js

```diff
@@ -4,17 +4,19 @@ const arr2d = [[1, 2], [3, 4], [5, 6]];
 const arr3d = [[[1, 2], [3, 4]], [[5, 6], [7, 8]]];
 
 function flattenArray2d(arr) {
-  // Replace this comment and the next line with your code
-  console.log(arr);
+  const newArray = [];
+  arr.map(arrFirst => arrFirst.map(arrSecond => newArray.push(arrSecond)));
+  return newArray;
 }
 
 function flattenArray3d(arr) {
-  // Replace this comment and the next line with your code
-  console.log(arr);
+  const newArray = [];
+  arr.map(newArr1 => newArr1.map(newArr2 => newArr2.map(newArr3 => newArray.push(newArr3))));
+  return newArray;
```

Comment:
Logic works fine but imagine if you have to do this for a 4-D or a 5-D array. What happens in that case? Suggestion: try reading about a programming concept called recursion and see if you can apply the same here. 🙂 

Also, have a look at `Array.prototype.flat` method which is newly introduced in MDN.

Reply:
Thak you for the advice, i rewite the code